typing/includemod.ml

```diff
@@ -587,7 +587,152 @@ let show_locs ppf (loc1, loc2) =
   show_loc "Expected declaration" ppf loc2;
   show_loc "Actual declaration" ppf loc1
 
-let include_err ppf = function
+let path_of_context = function
```

Comment:
I would find it more natural to have this and `context`, `context_mty`, `alt_context`... defined after the `Illegal_permutation` module rather than before. The idea is that, within an island of code that deals with the same concern, "interesting" / data-manipulating logic typically goes before printing/formatting logic.

Reply:
It is true that the position of the `Illegal_permutation` module was not ideal.